File: DSA/Stack/stack.java
package DSA;

import java.util.Stack;

public class stack {
    public static void insertatbottom(int data, Stack<Integer> s) {
        if (s.isEmpty()) {
            s.push(data);
            return;
        }
        int top = s.pop();
        insertatbottom(data, s);
        s.push(top);
    }

    public static void ReverseofStack(Stack<Integer> s) {
        Stack<Integer> reversed = new Stack<>();
        while (!s.isEmpty()) {
            int top = s.pop();
            reversed.push(top);
        }
        while (!reversed.isEmpty()) {
            System.out.println(reversed.peek());
            s.push(reversed.pop());
        }
    }

    public static void main(String[] args) {
        Stack<Integer> s = new Stack<>();
        s.push(9);
        s.push(2);
        s.push(3);
        s.push(4);
        ReverseofStack(s);
        insertatbottom(5, s);

        while (!s.isEmpty()) {
            System.out.println(s.peek());
            s.pop();
        }
    }
}

File: Hacktoberfest-2023/Programming/C/Data Structures/stack.c
#include <stdio.h>
#define length 5

int stack[length];
int tos =-1;

void push(int x){
	if(!isFull()){
		stack[++tos] = x;
	}
}

int pop(){
	if(!isEmpty())
		return stack[tos--];
}

int isFull(){
	if(tos >= length-1)
		return 1;
	return 0;
}

int isEmpty(){
	if(tos == -1)
		return 1;
	return 0;
}

void peek(){
	if(!isEmpty())
		printf("%d ", stack[tos]);
}

void display(){
	int i;
	
	for(i=tos; i >=0; i--){
		printf("%d ", stack[i]);	
	}
	
	printf("\n");
}

int main(){
	push(1);
	push(2);
	push(3);
	push(9);	
	display();
	pop();
	display();
	pop();
	display();
}
